qiling/extensions/tracing/formats/tenet.py

```diff
@@ -0,0 +1,103 @@
+# This code structure is copied and modified from the coverage extension
+
+import qiling
+from qiling.const import QL_ENDIAN
+from .base import QlBaseTrace
+from .registers import ArchRegs
+from unicorn.unicorn_const import UC_MEM_READ, UC_MEM_WRITE
+
+class QlDrTrace(QlBaseTrace):
+    """
+    Traces emulation and puts it into a format viewable in Tenet
+    IDAPro plugin Tenet: https://github.com/gaasedelen/tenet
+    """
+
+    FORMAT_NAME = "tenet"
+
+    def __init__(self, ql: qiling.Qiling):
+        super().__init__()
+        self.ql             = ql
+        self.deltas         = []
+        self.current_delta  = []
+        self.current_pc     = 0x0
+
+        self.arch_regs = ArchRegs(ql.arch)
+        self.register_values= dict()
+
+        # Initialize with ridiculous value so first delta isn't missed
+        for register in self.arch_regs.registers:
+            self.register_values[register] = 0xFEEDBABE
+
+
+    def _add_delta(self):
+        # Cover glitch cases where nothing changed
+        if self.current_delta != []:
+            # Join all delta fragments into delta line and append 
+            self.deltas.append(",".join(self.current_delta))
+            self.current_delta = []
+        return
+
+
+    @staticmethod
+    def mem_access_callback(ql, access, address, size, value, self):
+        access_type = None
+        # Set delta based on access type
+        if access == UC_MEM_READ:
+            # Since we are reading memory, we just read it ourselves
+            access_type = "mr"
+            value = ql.mem.read(address, size)
+            delta = f"{access_type}={hex(address)}:{value.hex()}"
+        elif access == UC_MEM_WRITE:
+            # Hook is before it's written, so we have to use the "value"
+            access_type = "mw"
+            if ql.arch.endian == QL_ENDIAN.EL:
```

Comment:
petty: Use ternary operation when an `if`-`else` construct asigns a value to the same variable.